### PR TITLE
Add CloudFront CDN to Discourse instances

### PR DESCRIPTION
**Pull request by flamingspaz**

### CODE REVIEW

File: modules/discourse/main.tf [hunk 1]
```diff
@@ -66,3 +69,14 @@ resource "aws_s3_bucket" "discourse-content" {
...
     }
 }
+
+module "discourse-cdn" {
+  source              = "git://github.com/mozilla/partinfra-terraform-cloudfrontssl.git"
+
+  origin_domain_name  = "${var.fqdn}"
+  origin_path         = "/"
+  origin_id           = "discourse-pull-origin"
+  alias               = "cdn-${var.environment}.discourse.mozilla-community.org"
```
Comment (johngian) [inline modules/discourse/main.tf]: Should we also handle the DNS changes in this PR?

File: modules/discourse/main.tf [hunk 2]
```diff
@@ -1,8 +1,11 @@
 variable "vpc_id" {}
+variable "aws_region" {}
 variable "discourse_elasticache_instance_size" {}
 variable "elasticache_subnet_group" {}
 variable "service_security_group_id" {}
 variable "environment" {}
+variable "fqdn" {}
+variable "ssl_certificates" { type = "map" }
```
Comment (johngian) [inline modules/discourse/main.tf]: Do you need to define ``map``?

File: discourse.tf [hunk 3]
```diff
@@ -16,4 +19,7 @@ module "discourse-staging" {
     discourse_elasticache_instance_size = "cache.t2.micro"
     service_security_group_id           = "${module.mesos-cluster-staging.mesos-cluster-slave-sg-id}"
     elasticache_subnet_group            = "${aws_elasticache_subnet_group.elasticache-staging-subnet-group.name}"
+    fqdn                                = "discourse.staging.paas.mozilla.community"
+    ssl_certificates                    = "${var.ssl_certificates}"
```
Comment (johngian) [inline discourse.tf]: You can always pass the specific certificate here instead of passing everything to the module.

File: discourse.tf [hunk 4]
```diff
@@ -6,6 +6,9 @@ module "discourse-production" {
     discourse_elasticache_instance_size = "cache.t2.medium"
     service_security_group_id           = "${module.mesos-cluster-production.mesos-cluster-slave-sg-id}"
     elasticache_subnet_group            = "${aws_elasticache_subnet_group.elasticache-production-subnet-group.name}"
+    fqdn                                = "discourse.mozilla-community.org"
+    ssl_certificates                    = "${var.ssl_certificates}"
```
Comment (johngian) [inline discourse.tf]: Same here.

File: modules/discourse/main.tf (hunk 1 above)
Comment (flamingspaz) [inline modules/discourse/main.tf]: We need to refactor DNS the same way I did SSL, so it's best to do this later with everything else.

**Pull request merged**